engine/runtime-compiler/src/main/java/org/enso/compiler/pass/analyse/PassPersistance.java

```diff
@@ -0,0 +1,155 @@
+package org.enso.compiler.pass.analyse;
+
+import java.io.IOException;
+import org.enso.compiler.core.Persistance;
+import org.enso.compiler.pass.analyse.AliasAnalysis.Graph;
+import org.enso.compiler.pass.resolve.DocumentationComments;
+import org.enso.compiler.pass.resolve.FullyQualifiedNames;
+import org.enso.compiler.pass.resolve.IgnoredBindings;
+import org.enso.compiler.pass.resolve.ModuleAnnotations;
+import org.enso.compiler.pass.resolve.TypeSignatures;
+import org.enso.interpreter.dsl.Persistable;
+import org.openide.util.lookup.ServiceProvider;
+import scala.Option;
+
+@Persistable(clazz = CachePreferenceAnalysis.WeightInfo.class, id = 1111)
+@Persistable(clazz = DataflowAnalysis.DependencyInfo.class, id = 1112)
+@Persistable(clazz = DataflowAnalysis.DependencyMapping.class, id = 1113)
+@Persistable(clazz = GatherDiagnostics.DiagnosticsMeta.class, id = 1114)
+@Persistable(clazz = DocumentationComments.Doc.class, id = 1115)
+@Persistable(clazz = AliasAnalysis$Info$Occurrence.class, id = 1116)
```

Comment:
Aaah, multi level Scala objects. My IDE cannot resolve these once again. Sad emoji :(

Reply:
I also don't like I have to use their "inner class qualified name".